tests/integration-tests/utils.py

```diff
@@ -38,37 +38,75 @@
 }
 
 
-class SetupError(BaseException):
-    """Exception to throw from infrastructure factory fixtures if the infrastructure setup fails"""
-
-    def __init__(self, *args, stack_events=None, cluster_details=None):
-        self.message = self._format_message(args[0], stack_events, cluster_details)
+def _format_stack_error(message, stack_events=None, cluster_details=None) -> str:
```

Comment:
We verify that this "message" is always a string
As we discuss we can also consider to add validation error here when cluster creation fail with validation

Reply:
Done.